src/grammar/ast.ts

```diff
@@ -320,6 +321,15 @@ export type AstStatementForEach = {
     loc: SrcInfo;
 };
 
+export type AstStatementDestruct = {
+    kind: "statement_destruct";
+    type: AstTypeId;
+    identifiers: Map<AstId, AstId>;
```

Comment:
hmmm, maps from `AstId` makes it sort of unusable (you cannot use it with `.get` methods and field descriptions)

Reply:
Yes, but you suggested using maps in the previous review. I thought about making it as `Map<string, string>` but it'll miss AST ids this way

Reply:
yeah, we should keep the location info, but also should make it usable as a map.
looks like a custom map-like data type with a `get` method that can work on strings could be used here